server/container/guice/blob-objectstorage-guice/src/main/java/org/apache/james/modules/objectstorage/ObjectStorageBlobConfiguration.java

```diff
@@ -143,6 +128,59 @@ public ReadyToBuild keystone3(SwiftKeystone3ObjectStorage.Configuration authConf
 
         public static class ReadyToBuild {
 
+            public static interface Factory {
+                static enum Factories {
+                    SWIFT("swift", new Swift()),
```

Comment:
you don't really need an instance (you don't need a class, neither), Factory being a functional interface, you can implement it with a single method.